.github/workflows/publish.yml

```diff
@@ -0,0 +1,30 @@
+name: Publish
+on:
+  push:
+    branches: master
+    paths: lib/rubocop/rspec/version.rb
+jobs:
+  publish:
+    name: Publish to RubyGems
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      id-token: write
```

Comment:
Are these permissions actually necessary?

Reply:
It is added because it was used in `example`. Is it unnecessary?
https://github.com/rubocop/rubocop-rspec/issues/1827#issue-2164094559
https://github.com/segiddins/rubygems-await/blob/136ee45fac859c913d7f4b745c63c9ecdc920e14/.github/workflows/push_gem.yml#L13-L15

Reply:
I ran a few tests and you are absolutely correct, they are both needed. I believe `id-token` is necessary in order to fetch the temporary auth token, and `contents` is necessary for pushing the git tag.